File: apps/api/app/Models/SignalRule.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class SignalRule extends Model
{
    use HasFactory;

    protected $fillable = [
        'rule_key',
        'name',
        'description',
        'scope_type',
        'chain_type',
        'signal_type',
        'default_direction',
        'horizon_label',
        'horizon_days',
        'min_signal_score',
        'enabled',
        'weight_profile',
        'trigger_conditions',
    ];

    protected $casts = [
        'enabled' => 'boolean',
        'weight_profile' => 'array',
        'trigger_conditions' => 'array',
        'min_signal_score' => 'decimal:2',
    ];

    public function signals(): HasMany
    {
        return $this->hasMany(TradingSignal::class, 'signal_rule_id');
    }
}

File: services/intelligence/app/models/signal_rule.py
from sqlalchemy import JSON, Boolean, DateTime, Numeric, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class SignalRule(Base):
    __tablename__ = "signal_rules"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    rule_key: Mapped[str] = mapped_column(String(64), unique=True, nullable=False)
    name: Mapped[str] = mapped_column(String(128), nullable=False)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    scope_type: Mapped[str] = mapped_column(String(32), nullable=False, default="event_chain")
    chain_type: Mapped[str | None] = mapped_column(String(64), nullable=True)
    signal_type: Mapped[str] = mapped_column(String(64), nullable=False)
    default_direction: Mapped[str] = mapped_column(String(16), nullable=False)
    horizon_label: Mapped[str] = mapped_column(String(32), nullable=False)
    horizon_days: Mapped[int] = mapped_column(nullable=False, default=5)
    min_signal_score: Mapped[float] = mapped_column(Numeric(5, 2), nullable=False, default=60)
    enabled: Mapped[bool] = mapped_column(Boolean, nullable=False, default=True)
    weight_profile: Mapped[dict | None] = mapped_column(JSON, nullable=True)
    trigger_conditions: Mapped[dict | None] = mapped_column(JSON, nullable=True)
    created_at: Mapped[DateTime | None] = mapped_column(DateTime, nullable=True)
    updated_at: Mapped[DateTime | None] = mapped_column(DateTime, nullable=True)
